### ISSUE TITLE
DarkThemeToggle does not have accessible name

### ISSUE THREAD

**Issue by tulup-conner** [themesberg/flowbite-react#106]
**Describe the bug**
The `<DarkThemeToggle />` has no accessible name, which means screen readers read it simply as, "Button", with no context.

**To Reproduce**
1. Add a `<DarkThemeToggle />` to your page
2. Focus the button in a screen reader

**Expected behavior**
An accessible name should be read aloud, like, "Toggle dark mode".

Alternatively, the button can be labelled as hidden for screen readers. I'm not sure, are there users who would have a screen reader, but can still discern dark/light modes?

**Screenshots**
n/a

**Additional context**
I've tried this on Apple VoiceOver.
